server/connector.go

```diff
@@ -189,6 +191,26 @@ func (c *connectorImpl) findAndConnectBackend(ctx context.Context, frontendConn
 	c.metrics.ActiveConnections.Add(1)
 	defer c.metrics.ActiveConnections.Add(-1)
 
+	// PROXY protocol implementation
+
+	remoteAddr, err := net.ResolveIPAddr("ip", strings.Split(backendHostPort, ":")[0])
+	if err != nil {
+		// handle error
+	}
+
+	header := &proxyproto.Header{
+		Version:           2,
+		Command:           proxyproto.PROXY,
+		TransportProtocol: proxyproto.TCPv4,
+		SourceAddr:        clientAddr,
+		DestinationAddr:   remoteAddr,
+	}
+
+	_, err = header.WriteTo(backendConn)
+	if err != nil {
+		// handle error
+	}
```

Comment:
Add a new command-line/env field to

https://github.com/itzg/mc-router/blob/8b54fdfb863bd2af7bbfb4fc38a750b311bdf34b/cmd/mc-router/main.go#L31-L44

such as `UseProxyProtocol` and then wrap this in a conditional block. The default would be false, keeping the behavior to existing users of this.

Reply:
FYI, documentation on that flags syntax:

https://pkg.go.dev/github.com/itzg/go-flagsfiller#section-documentation

Reply:
Tbh, I literally forgot to do that, in my mind I said that I should do that 😅.

I'll do in a few hours, when I'll be at my PC.

Reply:
implemented this in commit #6f4272d :D